python/geom.py

```diff
@@ -0,0 +1,173 @@
+import math
+import meep as mp
+
+
+def non_negative(x):
+    return x >= 0
+
+
+class Vector3(object):
+
+    def __init__(self, x=0.0, y=0.0, z=0.0):
+        self.x = float(x)
+        self.y = float(y)
+        self.z = float(z)
+
+    def __eq__(self, other):
+        return self.x == other.x and self.y == other.y and self.z == other.z
```

Comment:
Eventually, we'll want the usual vector addition, subtraction, `* scalar`, etcetera operations, but that can be another PR.